ape_infura/provider.py

```diff
@@ -34,6 +35,23 @@ def __init__(self):
 
 class Infura(Web3Provider, UpstreamProvider):
     network_uris: dict[tuple[str, str], str] = {}
+    api_keys: list[str] = []
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.load_api_keys()
+
+    def load_api_keys(self):
+        self.api_keys = []
+        for env_var_name in _ENVIRONMENT_VARIABLE_NAMES:
+            if env_var := os.environ.get(env_var_name):
+                self.api_keys.extend([key.strip() for key in env_var.split(",")])
+
+        if not self.api_keys:
+            raise MissingProjectKeyError()
+
+    def get_random_api_key(self) -> str:
```

Comment:
idea: have like a `reconnect` method (or just call disconnect and the connect()` in which it will change the api key instead of making it easy.

i dont see why we need to "get" the api keys i guess